app/src/main/java/kr/ac/kaist/team888/util/bezier/ParametricPolynomialCurve.java

```diff
@@ -0,0 +1,113 @@
+package kr.ac.kaist.team888.util.bezier;
+
+import org.apache.commons.math3.analysis.polynomials.PolynomialFunction;
+import org.apache.commons.math3.exception.DimensionMismatchException;
+import org.apache.commons.math3.exception.NullArgumentException;
+import org.apache.commons.math3.exception.OutOfRangeException;
+import org.apache.commons.math3.geometry.euclidean.twod.Vector2D;
+import org.apache.commons.math3.util.MathUtils;
+
+/**
+ * Parametric curve in 2D plane represented by two polynomials of time from 0 to 1.
+ *
+ * <p>The general form of this curve C is C(t)=[sum a_i t^i, sum b_i t^i]
+ * where 0≤t≤1 and a_i, b_i are real numbers. Thus, C is the function from [0,1] to R^2.
+ */
+public class ParametricPolynomialCurve extends ParametricCurve {
+  protected PolynomialFunction[] polynomials;
+
+  protected ParametricPolynomialCurve() {
+  }
+
+  /**
+   * Creates a new parametric polynomial curve with given polynomials.
+   *
+   * <p>This restricts to receive exactly two polynomials, one is for x-axis
+   * and other is for y-axis.
+   *
+   * @param polynomials an array of two polynomials for x-axis and y-axis, respectively
+   * @throws DimensionMismatchException if number of polynomials is not 2
+   * @throws NullArgumentException if either the array of polynomials
+   *     or one of polynomial is `null`
+   */
+  public ParametricPolynomialCurve(PolynomialFunction[] polynomials)
+      throws DimensionMismatchException, NullArgumentException {
+    MathUtils.checkNotNull(polynomials);
+    if (polynomials.length != 2) {
+      throw new DimensionMismatchException(polynomials.length, 2);
+    }
+    for (int i = 0; i < polynomials.length; i++) {
+      MathUtils.checkNotNull(polynomials[i]);
+    }
+    this.polynomials = polynomials;
+  }
+
+  /**
+   * Returns the array of polynomials.
+   *
+   * <p>This methods copies the array and returns it.
+   *
+   * @return the array of polynomials
+   */
+  public PolynomialFunction[] getPolynomials() {
+    return new PolynomialFunction[]{
+        new PolynomialFunction(polynomials[0].getCoefficients()),
+        new PolynomialFunction(polynomials[1].getCoefficients())
+    };
+  }
+
+  /**
+   * Returns the value at the given parameter.
+   *
+   * @param time parameter to compute the value within 0 to 1
+   * @return the value of the function
+   * @throws OutOfRangeException if the given parameter is not within 0 to 1
+   */
+  @Override
+  public double[] value(double time) throws OutOfRangeException {
```

Comment:
Do we need OutOfRangeException? How about mapping time less than 0 to 0 and more than 1 to 1.